tests/test_cases/test_cocotb/test_edge_triggers.py

```diff
@@ -233,3 +233,39 @@ async def wait_falling_edge():
     falling_coro = cocotb.fork(wait_falling_edge())
     cocotb.fork(Clock(dut.clk, 10, units='ns').start())
     await Combine(rising_coro, falling_coro)
+
+
+@cocotb.test()
+async def test_edge_on_vector(dut):
+    """Test that Edge() triggers on any 0/1 change in a vector"""
+
+    cocotb.fork(Clock(dut.clk, 100, "ns").start())
+
+    edge_cnt = 0
+    edge_cnt_exp = 0
+
+    async def wait_edge():
+        nonlocal edge_cnt
+        while True:
+            await Edge(dut.stream_out_data_registered)
+            edge_cnt = edge_cnt + 1
+
+    cocotb.fork(wait_edge())
+
+    dut.stream_in_data <= 0
+    await ClockCycles(dut.clk, 1)
+
+    for val in range(1, 2**len(dut.stream_in_data)-1):
+        # produce an edge by setting a value != 0:
+        dut.stream_in_data <= val
+        await ClockCycles(dut.clk, 1)
```

Comment:
Is there a reason to use `ClockCycles(clk, 1)` over `RisingEdge(clk)`?

Reply:
This is single-step driving the clock, not reacting to it.
A small side-effect is that I can test Xcelium which otherwise would fail because of #1076.

Reply:
(Previous comment was BS...)
No, no reason, will change.